feat: configuration reading from environment

ICavlek opened the pull request:
**Describe the changes**
Reading of configuration is now done from the environment instead of `config.toml`. Additionally, all variables are `base64` encoded.

**Related Issue(s)**
[issue](https://github.com/eigerco/axelar-solana-relayer/issues/45)

**Checklist**

- [ ] Tests have been added/updated for the changes.
- [ ] Documentation has been updated for the changes (if applicable).
- [ ] The code follows Rust's style guidelines.

ICavlek commented:
This is the initial proposition of the changes. For now, I have only removed variable `storage_path` from the `config.toml` for the showcase. Additionally, bridge struct `ConfigToDelete` is the one that reads `config.toml` without the `storage_path` variable.

Running the following:
```
STORAGE_PATH=Li9zdG9yYWdl cargo run --bin solana-axelar-relayer
```

Prints out:
```
./storage
```

I didn't go for complete removal of `Config` struct, because there are other repositories which depend on this.

Please comment and suggest if these proposed changes are sufficient so that I could continue.

eloylp reviewed (commented):
Hey @ICavlek thanks for giving this draft PR for expectatives alignment :)  Some points to review under my POV:
 * All variables are base64 encoded - I don't see this necessary, only for the ones which are multiline information or have special characters that could break the env var passing through
 * The dependency with the Config struct - Maybe we can make use of Clap for parsing env vars. It has a `env` [feature](https://docs.rs/clap/latest/clap/_features/index.html) for that. So we could hydrate the Config struct from env vars, leaving the rest of the code intact.
 
Let me know how this matched with your current context i might be missing something :)

ICavlek commented:
@eloylp I have added some updates:

- `env` parsing from `clap` - check in `rest_service::Config`
- `dotenv` - added reading of environment variables from `.env`, priorities have variables that are passed via command line. Easier for me to track down things, can be removed before merging.

For the `base64` encoding, ones that seem to me that are candidates for this are:

- `amplifier_component.identity` - that huge certificate that contains private key
- `solana_gateway_task_processor.signing_keypair` - big vector of `ints`
- `solana_listener_component.solana_ws` - contains api key
- `solana_rpc.solana_http_rpc` - contains api key

Others seem pretty fine to me and should be properly parsed without `base64` encoding.

eloylp commented:
> @eloylp I have added some updates:
> 
> * `env` parsing from `clap` - check in `rest_service::Config`
> * `dotenv` - added reading of environment variables from `.env`, priorities have variables that are passed via command line. Easier for me to track down things, can be removed before merging.
> 
> For the `base64` encoding, ones that seem to me that are candidates for this are:
> 
> * `amplifier_component.identity` - that huge certificate that contains private key
> * `solana_gateway_task_processor.signing_keypair` - big vector of `ints`
> * `solana_listener_component.solana_ws` - contains api key
> * `solana_rpc.solana_http_rpc` - contains api key
> 
> Others seem pretty fine to me and should be properly parsed without `base64` encoding.

Unless the api keys are very big or have special chars i would let them raw. for the rest looks good to me ! :)

ICavlek commented:
Ready for review:

`signing_keypair` 
- changed from `Keypair` to `String` because it doesn't implement `Clone` trait and it cannot be parsed from `env`
- `signing_keypair()` function that converts  `String` to`Keypair` type
- `base58` type as `String` from environment, cannot be encoded to `base64`
- removed tests that take byte array as input - not practical

`.env`
- adding this file can make it easier to define all environment variables there
- all except `AMPLIFIER_API_IDENTITY` can be properly parsed, too big

`AMPLIFIER_API_IDENTITY`
- has to be passed in via console as `AMPLIFIER_API_IDENTITY=<huge_value> cargo run --bin solana-axelar-relayer`

eloylp reviewed (approved):
LGTM 🚀

eloylp commented:
A friendly bump for reviewing this one @vaporif :)
